shared.go

```diff
@@ -131,6 +131,7 @@ type rawHeader struct {
 	Jwk   *JSONWebKey          `json:"jwk,omitempty"`
 	Kid   string               `json:"kid,omitempty"`
 	Nonce string               `json:"nonce,omitempty"`
+	Tpe   string               `json:"tpe,omitempty"`
```

Comment:
Shouldn't this be `typ` instead of `tpe`? 

See: https://tools.ietf.org/html/draft-ietf-oauth-json-web-token-32#section-5.1